File: PHP/string_palindrome.php
<?php # string_palindrome.php - Nate Nasteff 2020

/*
** A very simple set of functions, one that returns a boolean 
** after comparing the original string to the reversed string,
** and another function which takes said boolean as arg and
** tells the user if the word is a palindrome or not.
*/

$palindrome = function ($str) {
  echo "You entered: " . $str;
  $str = strtolower($str);
  return strrev($str) == $str ?
    true : false;
};

$is_palindrome = function ($bool) {
  return ($bool) ?
    "\nThis word is a palindrome.\n\n" : "\nThis word is not a palindrome.\n\n";
};

$str = "Milk";
$str2 = "Racecar";

echo $is_palindrome($palindrome($str));
echo $is_palindrome($palindrome($str2));


/*
**
** Sample I/O
**
** INPUT: "Milk" "Racecar"
**
** Output:
** 
** You entered: Milk
** This word is not a palindrome.
**
** You entered: Racecar
** This word is a palindrome.
**
**
** Time complexity: O(n)
** Space complexity: O(1)
**
*/

File: C/String_Palindrome.c
#include<stdio.h>
#include<string.h>

int main(){
    char str[50];
    printf("Enter any string: ");
    scanf("%s",str);

    int l = 0, h= strlen(str)-1;
    while(l<h){
        if(str[l++] != str[h--]){
            printf("The given string is not a palindrome");
            return 0;
        }
    }
    printf("The given string is a palindrome");
    return 0;
}

/* Problem name: To check whether a string is palindrome or not 

   1)
   Input: 
   Enter any string: palaalap

   Output:
   The given string is a palindrome

   2)
   Input:
   Enter any string: hey

   Output:
   The given string is not a palindrome 

   
   Time Complexity: O(n)
   Space Complexity: O(n)


*/
